sdks/java/harness/src/main/java/org/apache/beam/fn/harness/data/PCollectionConsumerRegistry.java

```diff
@@ -209,8 +208,7 @@ public FnDataReceiver<WindowedValue<?>> getMultiplexingConsumer(String pCollecti
             come up in the existing SDF expansion, but might be useful to support fused SDF nodes.
             This would require dedicated delivery of the split results to each of the consumers
             separately. */
-            return new MultiplexingMetricTrackingFnDataReceiver(
-                pcId, coder, ImmutableList.copyOf(consumerAndMetadatas));
+            return new MultiplexingMetricTrackingFnDataReceiver(pcId, coder, consumerAndMetadatas);
```

Comment:
This is good less copies of data.

Reply:
yes, we also already handle the case to not allow mutation after as well